### DISCUSSION TITLE
Querying PostgreSQL objects

### DISCUSSION THREAD

**Discussion by marek22k** [jeremyevans/sequel#2058] [General]
Hello,

I want to formulate a query in Sequel to get all rows that contain a subnet in the `prefix` column that is equal to or contains the specified one.
```sql
SELECT * FROM public.prefixorigins4
WHERE prefix  && inet '172.22.149.114/31'
```

Furthermore, I would like to make a query so that I get all subnets that have a certain subnet mask.
```sql
SELECT * FROM public.prefixorigins4
WHERE  masklen(prefix) = 24
```

Does anyone know how to do this? I have to admit that I'm a little bit worn out.

**Comment by jeremyevans** [comment]
```ruby
Sequel.extension :pg_inet_ops
ds = DB.from{public[:prefixorigins4]}
ds.where{Sequel.pg_inet_op(:prefix).contains_or_contained_by(inet('172.22.149.114/31'))}
ds.where{{masklen(:prefix) => 24}}
```

**Reply by marek22k** [reply]
Thank you for your answer! How would you express the operator ">>="? Is there a documentation on how these operators are expressed in Sequel?

**Reply by jeremyevans** [reply]
Sequel has a `pg_inet_ops` extension that implements inet functions and operators (including support for `>>=`): https://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/pg_inet_ops_rb.html
